src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -212,6 +227,11 @@ public void onError(Throwable e) {
         @Override
         public void onCompleted() {
             if (once.compareAndSet(0, 1)) {
+                // signal the production count at once instead of one by one
+                long c = produced;
+                if (c != 0L) {
```

Comment:
I think @davidmoten is right here, I don't see why you need to capture `produced`.